python/lsst/ts/eas/diurnal_timer.py

```diff
@@ -188,85 +187,153 @@ def __init__(self, sun_altitude: str | float = -18.0):
 
         self.sun_altitude = sun_altitude
         self.twilight_time: Time | None = None
+        self.sunrise_time: Time | None = None
 
         self._noon_loop_ready = asyncio.Event()
         self._twilight_loop_ready = asyncio.Event()
+        self._sunrise_loop_ready = asyncio.Event()
 
         self.noon_condition = asyncio.Condition()
         self.twilight_condition = asyncio.Condition()
+        self.sunrise_condition = asyncio.Condition()
 
         self.is_running = False
         self._tasks: list[asyncio.Task] = []
 
-    async def _run_noon_loop(self) -> None:
-        """Notifies noon_condition whenever it's noon.
+    @staticmethod
+    def _seconds_until(target: Time, now: Time | None = None) -> float:
+        """Returns time from `now` until `target` in seconds.
```

Comment:
Please use an active voice in doc strings so `Returns` -> `Return`. Do this from now on as this has been explained in the TSSW developer guidelines.